ggshield/verticals/secret/output/secret_json_output_handler.py

```diff
@@ -109,6 +110,36 @@ def flattened_policy_break(
             flattened_dict["incident_url"] = policy_breaks[0].incident_url
 
         for policy_break in policy_breaks:
-            flattened_dict["occurrences"].extend(policy_break.matches)
+            flattened_dict["occurrences"].extend(self.flatten_matches(policy_break))
 
         return flattened_dict
+
+    def flatten_matches(
+        self,
+        policy_break: PolicyBreak,
+    ) -> List[Dict[str, Any]]:
+        """
+        flatten_policy_breaks_by_line turns a list of policy breaks into a list of
+        tuples of (line, list of matches) mapping a line to a list of matches starting
+        at that line. The list is sorted by line number.
+        """
+        flattened_dict: List[Dict[str, Any]] = []
+        for match in policy_break.matches:
+            assert isinstance(match, ExtendedMatch)
+
+            flattened_match_dict: Dict[str, Any] = {
+                "match": match.match,
+                "type": match.match_type,
+                "line_start": match.line_start,
+                "line_end": match.line_end,
+                "index_start": match.span.column_index_start,
+                "index_end": match.span.column_index_end,
+            }
+            if match.pre_line_start is not None and match.pre_line_end is not None:
+                flattened_match_dict["pre_line_start"] = match.pre_line_start
+                flattened_match_dict["pre_line_end"] = match.pre_line_end
+            if match.post_line_start is not None and match.post_line_end is not None:
+                flattened_match_dict["post_line_start"] = match.post_line_start
+                flattened_match_dict["post_line_end"] = match.post_line_end
```

Comment:
I think the CI is unhappy because the `{pre|post}_line_{start|end}` are not always present.

Reply:
The CI was unhappy for two reasons : 
1. carriage return on windows
2. MatchSchema removed the key "type" as the expected key was "match_type". 

Regarding `{pre|post}_line_{start|end}`, I thought it was more logical to consistently set them to `None` when scanning a file and not a patch, as the notion of pre/post-commit doesn't exist in this context. This does change the actual behavior of ggshield however. If we don't want to do it, I can simply revert the changes on extended_match.py :smile: